core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,7 +64,23 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // Evaluate partitions in parallel. Partitions of each rdd will be cached by the `partitions`
+  // val in `RDD`.
+  private[spark] lazy val parallelPartitionEval: Boolean = {
+    val threshold = conf.getInt("spark.rdd.parallelListingThreshold", 10)
+    if (threshold > 0 && rdds.length > threshold) {
+      val parArray = rdds.toParArray
+      parArray.tasksupport = new ForkJoinTaskSupport(
+        new ForkJoinPool(math.min(threshold, 8)))
```

Comment:
The current code allows the user to turn parallelism off (thresh=0) or to control when parallel processing happens, but limits execution to a maximum of 8 threads. I think that's pretty reasonable.

Your suggestion allows setting the number of threads, but not controlling when execution happens in parallel. But controlling the number of threads is not the important part. At a minimum, users should be able to keep the current behavior.

If users need to set the number of threads, we can do that in a later patch when necessary. Otherwise, I think the current patch is the right way to go.